src/user-post/controller/post.controller.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  FileTypeValidator,
+  Get,
+  MaxFileSizeValidator,
+  Param,
+  ParseFilePipe,
+  Patch,
+  Post,
+  Req,
+  UploadedFile,
+  UseInterceptors,
+} from '@nestjs/common';
+import { EditPostDto, PostDto } from '../dto/post.dto/post.dto';
+import { PostService } from '../service/post.service';
+import { Posts } from '../entities/Post.entity';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { Request } from 'express';
+import { ApiTags } from '@nestjs/swagger';
+
+@Controller('post')
+@ApiTags('Post')
+export class PostController {
+  constructor(private postService: PostService) {}
+  @Post('')
+  @UseInterceptors(FileInterceptor('photo'))
+  create(
+    @Body() postFormData: PostDto,
+    @UploadedFile(
+      new ParseFilePipe({
+        fileIsRequired: false,
+        validators: [
+          new MaxFileSizeValidator({ maxSize: 10000000 }),
+          new FileTypeValidator({ fileType: '.(png|jpeg|jpg)' }),
+        ],
+      }),
+    )
+    photo: Express.Multer.File,
+  ): Promise<Posts[]> {
+    return this.postService.createPost(postFormData, photo);
+  }
+  @Get('user/:userId')
+  getPosts(@Param('userId') userId: string): Promise<Posts[]> {
+    return this.postService.getPosts(userId);
+  }
+  @Get('/getAllPosts')
+  getAllPosts(): Promise<Posts[]> {
+    console.log('...get all posts');
+    return this.postService.getAllPosts();
```

Comment:
All? Maybe pagination?

Reply:
What'll be if we'll a big database like facebook 😁

Reply:
The function 'getAllPosts' actually doesn't retrieve all the posts; it only fetches the first 5 of them. It's my mistake for naming it like that, which can be confusing. I know that you've probably noticed it's only getting 5 posts, and you may want me to rename the function. In this case, the name doesn't really make sense sorry 😅.